azure-common/azure/common/client_factory.py

```diff
@@ -25,6 +26,8 @@ def _instantiate_client(client_class, **kwargs):
     args = get_arg_spec(client_class.__init__).args
     if 'subscription_id' not in args:
         del kwargs['subscription_id']
+    elif sys.version_info < (3, 0) and isinstance( kwargs['subscription_id'], unicode ):
+        kwargs['subscription_id'] = str( kwargs['subscription_id'] )
```

Comment:
If you're going to be casting text down to `str` from `unicode` in Python 2.7 then you should encode with an explicit encoding. There's also unnecessary spaces within the parentheses.